### PR TITLE
ESLint config dependencies upgrade

### LINKED ISSUE #657: Upgrade dependencies of eslint-config-falcon
**What should be changed/improved?**
The dependencies of `eslint-config-falcon` are outdated and depend on ESLint 5, while the config itself depends on ESLint 6.

When simply doing `yarn add -D @deity/eslint-config-falcon eslint`, we get the following output:
```
warning "@deity/eslint-config-falcon > @typescript-eslint/eslint-plugin@1.13.0" has incorrect peer dependency "eslint@^5.0.0".
warning "@deity/eslint-config-falcon > @typescript-eslint/parser@1.13.0" has incorrect peer dependency "eslint@^5.0.0".
warning "@deity/eslint-config-falcon > eslint-config-airbnb@17.1.1" has incorrect peer dependency "eslint@^4.19.1 || ^5.3.0".
warning "@deity/eslint-config-falcon > eslint-config-airbnb > eslint-config-airbnb-base@13.2.0" has incorrect peer dependency "eslint@^4.19.1 || ^5.3.0".
```

So we indirectly depend on ESLint 5.

If we instead do `yarn add -D @deity/eslint-config-falcon eslint@^5.3.0`, we get the following:
```
warning " > @deity/eslint-config-falcon@0.1.4" has incorrect peer dependency "eslint@>=6.1.0".
```

Our own config depends on ESLint 6.

This presents a conflict where we always have some unmet peer dependency (cannot be both `^5.3.0` and `>=6.1.0`).

**Your suggested solution**

Upgrade the packages `@typescript-eslint/eslint-plugin`, `@typescript-eslint/parser` and `eslint-config-airbnb` to their latest versions, which support ESLint 6.

### LINKED ISSUE #736: ESLint error: "Definition for rule 'import-name' was not found"
**Describe the bug**
The ESLint config `@deity/eslint-config-falcon` produces a lint error on every `.ts` file, with message: "Definition for rule 'import-name' was not found".

**To Reproduce**
1. `yarn add -D eslint @deity/eslint-config-falcon`
2. Create `.eslintrc` with content:
```
{
    "extends": ["@deity/eslint-config-falcon"]
}
```
3. `touch index.ts`
4. `npx eslint index.ts`
5. Observe the error.

**Expected behavior**
There should not be an error about an undefined rule.

### PR DESCRIPTION
**Pull request by nimobeeren**
### Please check if the PR fulfills these requirements ###

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Changelog have been added / updated (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Dependency upgrade and some config changes.

**What is the current behavior? (You can also link to an open issue here)**
The dependencies of `eslint-config-falcon` are outdated, which causes errors (fix #657, fix #736).

**What is the new behavior (if this is a feature change)?**
With the upgrade of `eslint-config-airbnb`, some new rules were added. Some of these I decided to include, others were disabled:

Disabled new rules:
 - react/jsx-props-no-spreading
 - react/static-property-placement
 - react/jsx-curly-newline (conflicts with Prettier)

Added new rules with no code changes:
 - function-call-argument-newline
 - jsx-a11y/control-has-associated-label
 - react/prefer-read-only-props

Applied fixes for new rules:
 - prefer-object-spread
 - react/sort-comp
 - react/state-in-constructor
 - react/jsx-fragments
 - max-classes-per-file

Removed rules:
- id-length

Also I fixed an issue where Prettier and ESLint were conflicting, this is due to the fact that not all ESLint plugins were installed properly. They should be listed as peer dependencies of `eslint-config-falcon`, similar to [airbnb package.json](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/package.json). Otherwise ESLint (which is running on the workspace root) cannot find these plugins.

**Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)**

**Other information:**
Reviewers: please discuss about the changed rules. I disabled some rules that I didn't like or were too much work to change.

### CODE REVIEW

**Comment by nimobeeren** [comment]
Should we also implement #552 in this PR?

**Comment by EddyVinck** [comment]
I'm not seeing the [`function-call-argument-newline`](https://eslint.org/docs/rules/function-call-argument-newline) rule you mentioned. I hope it's not set to `"always"`, though :slightly_smiling_face:

**Comment by nimobeeren** [comment]
@EddyVinck I used `https://www.npmjs.com/package/eslint-find-rules` and it does list that rule as active. I wasn't able to trigger the rule using the examples from ESLint docs: either it was already OK or Prettier fixed it.

File: package.json [hunk 1]
```diff
@@ -19,18 +19,28 @@
     "lint": "eslint --ext js,ts,tsx ./"
   },
   "devDependencies": {
-    "@deity/eslint-config-falcon": "^1.0.0",
+    "@deity/eslint-config-falcon": "1.0.0",
```
Comment (mateuszwozniak) [inline package.json]: We use `^` for our packages so please revert that one.

File: packages/falcon-ui-kit/src/Product/ProductConfigurator.ts [hunk 2]
```diff
@@ -54,9 +54,12 @@ type ProductConfiguratorState = {
  * Currently only configurable options are supported, in the future that class will handle bundled products as well as custom product attributes.
  */
 export class ProductConfigurator extends React.Component<ProductConfiguratorProps, ProductConfiguratorState> {
-  state: ProductConfiguratorState = {
-    selectedConfigurableOptions: {}
-  };
+  constructor(props) {
```
Comment (mateuszwozniak) [inline packages/falcon-ui-kit/src/Product/ProductConfigurator.ts]: Do you remember why this has been changed? I personally prefer state outside the contstructor.

File: packages/falcon-front-kit/src/Routes/OnlyUnauthenticatedRoute.tsx [hunk 3]
```diff
@@ -9,15 +9,15 @@ export type OnlyUnauthenticatedRouteProps = {
...
-  static defaultProps = {
-    redirectTo: '/'
-  };
-
   static propTypes = {
     ...(Route as any).propTypes,
     redirectTo: PropTypes.string
   };
 
+  static defaultProps = {
```
Comment (mateuszwozniak) [inline packages/falcon-front-kit/src/Routes/OnlyUnauthenticatedRoute.tsx]: I assume reordering was forced by some rule, right? If so then I'd rather suggest to disable that rule as it's not always a best choice to reorder the methods just for the sake of making eslint happy ;)

**Pull request closed without merging**